tools/KeyfileIOUtils.cpp

```diff
@@ -64,6 +73,18 @@ static void validateRSAPrivateKey(const std::string& key) {
   if (!std::regex_match(key, std::regex("[0-9A-Fa-f]+"))) throw std::runtime_error("Invalid RSA private key: " + key);
 }
 
+static void validateEdDSAPublicKey(const std::string& key) {
+  if (!(key.length() == EdDSAPublicKeyByteSize * 2) && (std::regex_match(key, std::regex("[0-9A-Fa-f]+")))) {
```

Comment:
I've implemented this check in a general way which saves the duplication of the regex. Maybe it should be extracted to our new crypto utils:
see Cryptosystem::validateKey.

Reply:
New function created to validate keys: `bool isValidKey(const std::string& keyName, const std::string& key, size_t expectedSize)`.

`validateEdDSAPublicKey()`, `validateEdDSAPrivateKey()`, & `Cryptosystem::validateKey()` removed.